cmd/kubeadm/app/cmd/phases/certs.go

```diff
@@ -51,85 +51,17 @@ var (
-
-	etcdServerCertLongDesc = fmt.Sprintf(normalizer.LongDesc(`
-		Generates the etcd serving certificate and key and saves them into %s and %s files.
-		
-		The certificate includes default subject alternative names and additional SANs provided by the user;
-		default SANs are: localhost, 127.0.0.1.
-
-		If both files already exist, kubeadm skips the generation step and existing files will be used.
-		`+cmdutil.AlphaDisclaimer), kubeadmconstants.EtcdServerCertName, kubeadmconstants.EtcdServerKeyName)
-
-	etcdPeerCertLongDesc = fmt.Sprintf(normalizer.LongDesc(`
-		Generates the etcd peer certificate and key and saves them into %s and %s files.
-		
-		The certificate includes default subject alternative names and additional SANs provided by the user;
-		default SANs are: <node-name>, <apiserver-advertise-address>.
-
-		If both files already exist, kubeadm skips the generation step and existing files will be used.
-		`+cmdutil.AlphaDisclaimer), kubeadmconstants.EtcdPeerCertName, kubeadmconstants.EtcdPeerKeyName)
-
-	etcdHealthcheckClientCertLongDesc = fmt.Sprintf(normalizer.LongDesc(`
-		Generates the client certificate for liveness probes to healthcheck etcd and the respective key,
-		and saves them into %s and %s files.
-
-		If both files already exist, kubeadm skips the generation step and existing files will be used.
-		`+cmdutil.AlphaDisclaimer), kubeadmconstants.EtcdHealthcheckClientCertName, kubeadmconstants.EtcdHealthcheckClientKeyName)
-
-	apiServerEtcdServerCertLongDesc = fmt.Sprintf(normalizer.LongDesc(`
-		Generates the client certificate for the API server to connect to etcd securely and the respective key,
-		and saves them into %s and %s files.
-
-		If both files already exist, kubeadm skips the generation step and existing files will be used.
-		`+cmdutil.AlphaDisclaimer), kubeadmconstants.APIServerEtcdClientCertName, kubeadmconstants.APIServerEtcdClientKeyName)
-
 	saKeyLongDesc = fmt.Sprintf(normalizer.LongDesc(`
 		Generates the private key for signing service account tokens along with its public key, and saves them into
 		%s and %s files.
-
 		If both files already exist, kubeadm skips the generation step and existing files will be used.
 		`+cmdutil.AlphaDisclaimer), kubeadmconstants.ServiceAccountPrivateKeyName, kubeadmconstants.ServiceAccountPublicKeyName)
 
-	frontProxyCaCertLongDesc = fmt.Sprintf(normalizer.LongDesc(`
-		Generates the front proxy CA certificate and key and saves them into %s and %s files.
-
-		If both files already exist, kubeadm skips the generation step and existing files will be used.
-		`+cmdutil.AlphaDisclaimer), kubeadmconstants.FrontProxyCACertName, kubeadmconstants.FrontProxyCAKeyName)
-
-	frontProxyClientCertLongDesc = fmt.Sprintf(normalizer.LongDesc(`
-		Generates the front proxy client certificate and key and saves them into %s and %s files.
+	genericLongDesc = normalizer.LongDesc(`
+		Generates the %[1]s, and saves them into %[2]s.cert and %[2]s.key files.%[3]s
```

Comment:
what does `%[1]s` do?

Reply:
it's the first index. I added it to all of them because I use the second argument twice, and I thought it'd be more obvious what was happening this way.

Reply:
ohhhh this is part of the format string. ok i got it, thx. I've never come across this before